python/cudf/cudf/core/dataframe.py

```diff
@@ -341,7 +341,16 @@ def _setitem_tuple_arg(self, key, value):
                 )
             self._frame._data.insert(key[1], new_col)
         else:
-            if isinstance(value, (cupy.ndarray, np.ndarray)):
+            error_msg = (
+                "shape mismatch: value array of shape {value1} "
+                "could not be broadcast to indexing result of "
+                "shape {value2}"
+            )
+            if is_scalar(value) or isinstance(value, range):
```

Comment:
For `range`, I don't think this is correct. It should be treated as "1d-value" in else branch after L403. Can we also add a test to this?